### PR TITLE
New feature #18375: [security] Some plugin settings need encryption

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

**Comment by Shnoulle** [comment]
Yes, this fix the issue about API upgrade and password … 

I like to have a way to upgrade my plugin with existing password, i mean : with a plugin moving from version 1.2.4 to 1.3.0 with `New feature : encrypt password`

- Check current version (in DB) : if it's 1.2.4 : get the setting uncrypted
- Save it crypted 
- Update version number

I can do the same using a new setting ( current name : `smtppass` , new name : `smtpasscrypted`) delete previous one when goes to setting. 
Let me think for a few hours

**Comment by gabrieljenik** [comment]
I think this discussion should be followed on this ticket: https://bugs.limesurvey.org/view.php?id=9056

**Review by Shnoulle — APPROVED** [review]

**Comment by Shnoulle** [comment]
It work , previous value are not updated :)

**Review by Shnoulle — CHANGES_REQUESTED** [review]
The QuestionAttribute settings are not related to PluginSetting.

Must move to another file 

- https://manual.limesurvey.org/NewQuestionAttributes : get array, can append to array. Use QuestionAttribute model
- https://manual.limesurvey.org/NewSurveySettings : use (by default) PluginSetting model
- https://github.com/LimeSurvey/LimeSurvey/blob/e3efd7fb1771e39d18396315fa7b2be56b78e800/application/libraries/PluginManager/PluginBase.php#L257 use (by default) PluginSetting model

File: tests/data/plugins/NewQuestionAttributesPlugin.php [hunk 1]
```diff
@@ -4,6 +4,8 @@ class NewQuestionAttributesPlugin extends PluginBase
 {
     protected static $description = 'Dummy plugin for testing newQuestionAttributes event';
     protected static $name = 'NewQuestionAttributesPlugin';
+    protected $storage = 'DbStorage';
```
[the remark this answers is not in the document]
Comment (Shnoulle) [reply tests/data/plugins/NewQuestionAttributesPlugin.php:7]: For me NewQuestionAttributesPlugin test QuestionAttributes and not something else .

We can use same plugin, but not in same file , no ?

**Review by Shnoulle — APPROVED** [review]
:+1: 

Maybe adding newSurveySettings event after.

**Comment by gabrieljenik** [comment]
@c-schmitz @olleharstedt Not sure about the "need testing" tag here.
This is an automated test being added :)

**Pull request merged**